dbt_semantic_interfaces/type_enums/date_part.py

```diff
@@ -40,4 +40,4 @@ def to_int(self) -> int:
     @property
     def compatible_granularities(self) -> List[TimeGranularity]:
         """Granularities that can be queried with this date part."""
```

Comment:
This is so confusing, can we explain it here?

We are switching this to allow TimeGranularity.DAY for DatePart.MONTH, correct? So it'd now look like:

```
SELECT date_part(month, metric_time__day), count(*)
FROM ...
GROUP BY date_part(month, metric_time__day)
```

Whereas before we only allowed something like this:

```
SELECT date_part(day, metric_time__month), count(*)
FROM ...
GROUP BY date_part(day, metric_time__month)
```

Ok, that makes sense to me now.

Maybe we can document this as:

```
We support querying granularities that are the same underlying granularity or coarser than the requested date_part. This is because extracting the "day" part from a monthly granularity will always produce a date_part value of `1`, which is probably not what the user wants.
```

Reply:
@tlento this is just the dev release for a PR that already merged to main, you can see [here](https://github.com/dbt-labs/dbt-semantic-interfaces/pull/212).
This is not actually a behavior change for queries, this was only used in an error message.